portfolio/src/main/webapp/script.js

```diff
@@ -89,4 +93,60 @@ function deleteComments () {
       }
     });
   }
+}
+
+/**
+ * Updates and displays votes
+ */
+function updateVotes (id, vote) {
+  switch (vote) {
+    case "up":
+      if (document.getElementById(id).classList.contains("unvoted")) {
```

Comment:
I don't understand the logic behind the various if/else cases, can you please add a comment?

Reply:
Will do. I'm in the middle of improving this at the minute, hoping it'll be clearer by the time I'm finished :)